src/sentry/receivers/onboarding.py

```diff
@@ -404,22 +405,27 @@ def record_event_with_first_minified_stack_trace_for_project(project, event, **k
 
     # First, only enter this logic if we've never seen a minified stack trace before
     if not project.flags.has_minified_stack_trace:
-        # Next, attempt to update the flag, but ONLY if the flag is currently not set.
-        # The number of affected rows tells us whether we succeeded or not. If we didn't, then skip sending the event.
-        # This guarantees us that this analytics event will only be ever sent once.
-        affected = Project.objects.filter(
-            id=project.id, flags=F("flags").bitand(~Project.flags.has_minified_stack_trace)
-        ).update(flags=F("flags").bitor(Project.flags.has_minified_stack_trace))
-
-        if affected:
-            analytics.record(
-                "first_event_with_minified_stack_trace_for_project.sent",
-                user_id=user.id,
-                organization_id=project.organization_id,
-                project_id=project.id,
-                platform=event.platform,
-                url=dict(event.tags).get("url", None),
-            )
+
+        if project.date_added > START_DATE_TRACKING_FIRST_EVENT_WITH_MINIFIED_STACK_TRACE_PER_PROJ:
+            # Next, attempt to update the flag, but ONLY if the flag is currently not set.
+            # The number of affected rows tells us whether we succeeded or not. If we didn't, then skip sending the event.
+            # This guarantees us that this analytics event will only be ever sent once.
+            affected = Project.objects.filter(
+                id=project.id, flags=F("flags").bitand(~Project.flags.has_minified_stack_trace)
+            ).update(flags=F("flags").bitor(Project.flags.has_minified_stack_trace))
+
+            if affected:
+                analytics.record(
+                    "first_event_with_minified_stack_trace_for_project.sent",
+                    user_id=user.id,
+                    organization_id=project.organization_id,
+                    project_id=project.id,
+                    platform=event.platform,
+                    url=dict(event.tags).get("url", None),
+                )
+        else:
+            # If the project is older than the cutoff date, then we don't send analytics, but we still set the flag
+            project.update(flags=F("flags").bitor(Project.flags.has_minified_stack_trace))
```

Comment:
@scefali can you help me to understand how the bitor works? I am not sure how `false` or `true` is being applied

Reply:
nevermind... @RaduW explained this to me 😉